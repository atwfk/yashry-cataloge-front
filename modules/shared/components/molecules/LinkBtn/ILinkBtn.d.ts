declare namespace ILinkBtn {
  export interface IProps {
    path: string;
    active?: boolean;
    linkId: string;
  }
}

export { ILinkBtn };
