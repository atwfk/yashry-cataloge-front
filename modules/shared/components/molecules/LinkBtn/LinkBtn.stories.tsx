import type { ReactElement } from "react";
import type { Story } from "@storybook/react";
import LinkBtn from "./LinkBtn";
import type { ILinkBtn } from "./ILinkBtn";

const args = {
  title: "Molecules/LinkBtn",
  component: LinkBtn,
  argTypes: {
    active: {
      control: { type: "boolean" },
    },
  },
};

export default args;

const Template: Story<ILinkBtn.IProps> = (args): ReactElement => (
  <LinkBtn {...args}>normal text</LinkBtn>
);

export const linkBtn = Template.bind({});
linkBtn.args = {
  active: false,
  path: "",
  linkId: "1",
};
