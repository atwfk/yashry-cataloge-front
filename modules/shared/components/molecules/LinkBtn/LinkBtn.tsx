import React from "react";
import type { FC, ReactElement } from "react";
import Link from "next/link";
import P from "../../atoms/Paragraph/Paragraph";
import { ILinkBtn } from "./ILinkBtn";

const LinkBtn: FC<ILinkBtn.IProps> = ({
  path,
  children,
  active,
  linkId,
}): ReactElement => {
  return (
    <Link href={path}>
      <a id={linkId}>
        <P
          fontColor={active ? "accent" : "dark"}
          fontSize="md"
          fontWeight="normal"
        >
          {children}
        </P>
      </a>
    </Link>
  );
};

export default LinkBtn;
