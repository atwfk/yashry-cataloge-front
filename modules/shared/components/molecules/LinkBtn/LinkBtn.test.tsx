import React from "react";
import LinkBtn from "./LinkBtn";
import renderer from "react-test-renderer";

describe("LinkBtn", () => {
  it("should render link button with active={false}", () => {
    const tree = renderer
      .create(
        <LinkBtn path="/" linkId="1">
          normal text
        </LinkBtn>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render link button with active={true}", () => {
    const tree = renderer
      .create(
        <LinkBtn path="/" linkId="1" active>
          normal text
        </LinkBtn>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
