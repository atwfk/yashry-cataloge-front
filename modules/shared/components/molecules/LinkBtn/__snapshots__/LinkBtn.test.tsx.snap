// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkBtn should render link button with active={false} 1`] = `
<a
  href="/"
  id="1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <p
    className="text-base text-dark font-normal"
  >
    normal text
  </p>
</a>
`;

exports[`LinkBtn should render link button with active={true} 1`] = `
<a
  href="/"
  id="1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <p
    className="text-base text-accent font-normal"
  >
    normal text
  </p>
</a>
`;
